eventuate-core/src/test/scala/com/rbmhtechnology/eventuate/EventsourcedActorTestKit.scala

```diff
@@ -0,0 +1,295 @@
+    processEventlog(Some(Seq(expected)), max)
+  }
+
+
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  // ++++++++++++ cleaning one or multiple events +++++++++++
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+
+  /** Pops one event from event log without returning it */
+  def popFromEventLog(max: FiniteDuration = expectEventTimeOut) {
+    processEventlog(None, max)
+  }
+
+  /**
+    * Pops all events from event log. Can be used to simply clean current event log in order not to explicitly expect
+    * messages that are not of interest and furthermore to 'unblock' actor, which waits for [[WriteSuccess]] messages
+    * before doing anything but persisting again. <br>
+    *
+    * Note: handle with care. In order to be sure that sending events to logProbe is not blocked by waiting for
+    * WriteSuccess in logProbe (which happens if events to be persisted are stashed in persisting actor), logProbe
+    * has to be queried until no more messages can be expected (see processEventlog). Which means this takes at least
+    * max to terminate.
+    */
+  def cleanEventLog(max: FiniteDuration = expectEventTimeOut): Unit = {
+
+    while (Try(processEventlog(None, max)).isInstanceOf[Success[Seq[Any]]]) {
+
+    }
+  }
+
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  // +++++++++++ Querying events currently stashed ++++++++++
+  // ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+  /** Returns single event currently stashed in event log (which is than popped) */
+  def queryNextEvent(max: FiniteDuration = expectEventTimeOut) = processEventlog(None, max)
+
+  /**
+    * Cleans all events from event log. <br>
+    *
+    * Note: handle with care. In order to be sure that sending events to logProbe is not blocked by waiting for
+    * WriteSuccess in logProbe (which happens if events to be persisted are stashed in persisting actor), logProbe
+    * has to be queried until no more messages can be expected (see processEventlog). Which means this takes at least
+    * max to terminate.
+    *
+    * @param max
+    * @return
+    */
+  def queryAllEvents(max: FiniteDuration = expectEventTimeOut): Seq[Any] = {
+
+    var events: Seq[Any] = Seq.empty
+    var continue = true
+
+    while (continue) {
+      Try(processEventlog(None, max)) match {
+        case Success(nextEvents) =>
+          events = events ++ nextEvents
+        case Failure(_) =>
+          continue = false
+      }
+    }
+
+    events
+  }
+
+  /**
+    * Simulates Eventlog behaviour. Expects an [[Write]] on 'logProbe'. If 'expected' event is defined equality to
+    * written event(s) is asserted. Sender of event is told about [[WriteSuccess]] allowing him to continue processing.
+    *
+    * @return events that were contained by [[Write]]
+    */
+  private def processEventlog[T](expected: Option[Seq[T]], max: FiniteDuration = expectEventTimeOut) = {
```

Comment:
Maybe this method can be decomposed: 

- one method for writing and replying with `WriteSuccess`
- another for making assertions

The optional `expected` in the current version is a hint that such a decomposition is what you want